### PR TITLE
Community: Azure CosmosDB No Sql Vector Store: Full Text and Hybrid Search Support

### PR DESCRIPTION
**Pull request by aayush3011**
Thank you for contributing to LangChain!

- Added [full text](https://learn.microsoft.com/en-us/azure/cosmos-db/gen-ai/full-text-search) and [hybrid search](https://learn.microsoft.com/en-us/azure/cosmos-db/gen-ai/hybrid-search) support for Azure CosmosDB NoSql Vector Store
- Added a new enum called CosmosDBQueryType which supports the following values:
    - VECTOR = "vector"
    - FULL_TEXT_SEARCH = "full_text_search"
    - FULL_TEXT_RANK = "full_text_rank"
    - HYBRID = "hybrid"
- User now needs to provide this query_type to the similarity_search method for the vectorStore to make the correct query api call.
- Added a couple of work arounds as for the FULL_TEXT_RANK and HYBRID query functions we don't support parameterized queries right now. I have added TODO's in place, and will remove these work arounds by end of January.
- Added necessary test cases and updated the 

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Comment by efriis** [comment]
could you check the ci failures?

Also, if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}

Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/

would probably live out of https://github.com/langchain-ai/langchain-azure repo

**Comment by aayush3011** [comment]
@efriis, we are already working with you guys on porting all the azure integrations to the above repo you mentioned. Once everything is ported we would maintain the langchain-azure repo.

File: libs/community/langchain_community/vectorstores/azure_cosmos_db.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def __init__(
         index_name: str = "vectorSearchIndex",
         text_key: str = "textContent",
         embedding_key: str = "vectorContent",
-        application_name: str = "LANGCHAIN_PYTHON",
+        application_name: str = "LangChain-CDBMongoVCore-VectorStore-Python",
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db.py:85]:
```suggestion
        application_name: str = "LANGCHAIN_PYTHON",
```

File: libs/community/langchain_community/vectorstores/azure_cosmos_db.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def from_connection_string(
         connection_string: str,
         namespace: str,
         embedding: Embeddings,
-        application_name: str = "LANGCHAIN_PYTHON",
+        application_name: str = "LangChain-CDBMongoVCore-VectorStore-Python",
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db.py:124]:
```suggestion
        application_name: str = "LANGCHAIN_PYTHON",
```
these default changes are unfortunately likely breaking - will bring them back to the old one! When moving to the partner package, you're welcome to update

**Review by efriis — APPROVED** [review]

File: libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py [hunk 3]
```diff
@@ -34,9 +62,14 @@ def __init__(
...
         cosmos_container_properties: Dict[str, Any],
         cosmos_database_properties: Dict[str, Any],
+        full_text_policy: Optional[Dict[str, Any]] = None,
         database_name: str = "vectorSearchDB",
         container_name: str = "vectorSearchContainer",
+        text_key: str = "text",
+        embedding_key: str = "embedding",
+        metadata_key: str = "metadata",
         create_container: bool = True,
+        full_text_search_enabled: bool = False,
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:72]:
```suggestion
        database_name: str = "vectorSearchDB",
        container_name: str = "vectorSearchContainer",
        create_container: bool = True,
        *,
        full_text_policy: Optional[Dict[str, Any]] = None,
        text_key: str = "text",
        embedding_key: str = "embedding",
        metadata_key: str = "metadata",
        full_text_search_enabled: bool = False,
```
ordering change also breaking

File: libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py [hunk 4]
```diff
@@ -47,30 +80,42 @@ def __init__(
...
             indexing_policy: Indexing Policy for the container.
             cosmos_container_properties: Container Properties for the container.
             cosmos_database_properties: Database Properties for the container.
+            text_key: Text key to use for text property which will be
+                      embedded in the data schema.
+            embedding_key: Embedding key to use for vector embedding.
+            metadata_key: Metadata key to use for data schema.
+            create_container: Set to true if the container does not exist.
+            full_text_search_enabled: Set to true if the full text search is enabled.
         """
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:93]:
```suggestion
            indexing_policy: Indexing Policy for the container.
            cosmos_container_properties: Container Properties for the container.
            cosmos_database_properties: Database Properties for the container.
            full_text_policy: Full Text Policy for the container.
            text_key: Text key to use for text property which will be
                      embedded in the data schema.
            embedding_key: Embedding key to use for vector embedding.
            metadata_key: Metadata key to use for data schema.
            create_container: Set to true if the container does not exist.
            full_text_search_enabled: Set to true if the full text search is enabled.
        """
```

Comment (efriis) [reply libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:93]: going to do these from a branch there's a few of these

File: libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py (hunk 3 above)
Comment (efriis) [reply libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:72]: nvm these are kwargs

File: libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py (hunk 4 above)
Comment (efriis) [reply libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:93]: nvm these are kwargs

**Review by efriis — APPROVED** [review]

**Pull request merged**